pkg/expression/builtin_cast_vec.go

```diff
@@ -345,11 +345,18 @@ func (b *builtinCastDurationAsIntSig) vecEvalInt(ctx sessionctx.Context, input *
 
 		duration.Duration = ds[i]
 		duration.Fsp = fsp
-		dur, err := duration.RoundFrac(types.DefaultFsp, ctx.GetSessionVars().Location())
-		if err != nil {
-			return err
+
+		if b.tp.GetType() == mysql.TypeYear {
```

Comment:
I would recommend to use a local bool variable outside the for loop to do this check, so that the loop body can be tighter.